R-package/R/lgb.plot.tree.R

```diff
@@ -0,0 +1,184 @@
+        sapply(strsplit(as.character(x), "||", fixed = TRUE), function(el) {
+            any(el == "0")
+        })
+    }
+    dt[zero_present(Split), Missing := Yes]
+    # dt[, c('parent', 'default_left') := NULL]
+    # data.table::setcolorder(dt, c('Tree','Node','ID','Feature','decision_type','Split','Yes','No','Missing','Gain','Cover','Value'))
+    # create the label text
+    dt[, label := paste0(
+        Feature,
+        "\nCover: ", Cover,
+        ifelse(Feature == "Leaf", "", "\nGain: "), ifelse(Feature == "Leaf", "", round(Gain, 4)),
+        "\nValue: ", round(Value, 4)
+    )]
+    # style the nodes - same format as xgboost
+    dt[Node == 0, label := paste0("Tree ", Tree, "\n", label)]
+    dt[, shape := "rectangle"][Feature == "Leaf", shape := "oval"]
+    dt[, filledcolor := "Beige"][Feature == "Leaf", filledcolor := "Khaki"]
+    # in order to draw the first tree on top:
+    dt <- dt[order(-Tree)]
+    nodes <- DiagrammeR::create_node_df(
+        n         = nrow(dt),
+        ID        = dt$ID,
+        label     = dt$label,
+        fillcolor = dt$filledcolor,
+        shape     = dt$shape,
+        data      = dt$Feature,
+        fontcolor = "black"
+    )
+    # round the edge labels to 4 s.f. if they are numeric
+    # as otherwise get too many decimal places and the diagram looks bad
+    # would rather not use suppressWarnings
+    numeric_idx <- suppressWarnings(!is.na(as.numeric(dt[["Split"]])))
+    dt[numeric_idx, Split := round(as.numeric(Split), 4)]
+    # replace indices with feature levels if rules supplied
+    levels.to.names <- function(x, feature_name, rules) {
+        lvls <- sort(rules[[feature_name]])
+        result <- strsplit(x, "||", fixed = TRUE)
+        result <- lapply(result, as.numeric)
+        levels_to_names <- function(x) {
+            names(lvls)[as.numeric(x)]
+        }
+        result <- lapply(result, levels_to_names)
+        result <- lapply(result, paste, collapse = "\n")
+        result <- as.character(result)
+    }
+    if (!is.null(rules)) {
+        for (f in names(rules)) {
+            dt[Feature == f & decision_type == "==", Split := levels.to.names(Split, f, rules)]
+        }
+    }
+    # replace long split names with a message
+    dt[nchar(Split) > 500, Split := "Split too long to render"]
+    # create the edge labels
+    edges <- DiagrammeR::create_edge_df(
+        from = match(dt[Feature != "Leaf", c(ID)] %>% rep(2), dt$ID),
+        to = match(dt[Feature != "Leaf", c(Yes, No)], dt$ID),
+        label = dt[Feature != "Leaf", paste(decision_type, Split)] %>%
+            c(rep("", nrow(dt[Feature != "Leaf"]))),
+        style = dt[Feature != "Leaf", ifelse(Missing == Yes, "bold", "solid")] %>%
+            c(dt[Feature != "Leaf", ifelse(Missing == No, "bold", "solid")]),
+        rel = "leading_to"
+    )
+    # create the graph
+    graph <- DiagrammeR::create_graph(
+        nodes_df = nodes,
+        edges_df = edges,
+        attr_theme = NULL
+    ) %>%
```

Comment:
In this project, by convention we:

* do not use the `%>%` operator
* use comma-first style everywhere

Please update this code and all the other code you're adding to follow that. Keeping all of the code looking the same across the codebase helps us to develop and review changes.

Reply:
`xgboost`'s implementation of similar functionality might be useful as a reference. See https://github.com/dmlc/xgboost/blob/e988b7cf1515b08ad0f949c26beb043ce0b33fe8/R-package/R/xgb.plot.tree.R#L159-L181